frontend/components/docs/js/docs.component.ts

```diff
@@ -59,6 +59,25 @@ class DocsController implements ng.IController {
 	}
 
 	public watchers() {
+
+		this.$scope.$watch(() => {
+			return this.ViewerService.selection;
```

Comment:
Not sure I entirely follow your thought process here.

To my understanding the viewer receives a callback from the viewer itself (never touching the TreeService as far as I can tell), which in turn is handled by ViewerService. Is there something I'm missing about how we get the Model/Account/ID?

Here we used to call DocsService directly (coupling DocsService to the ViewerService and also causing a circular dependency). Now we just watch the ViewerService from the component waiting for information on new selections, and call the DocsService from the component (gluing together services rather than coupling them). In short all we've really done here is decoupled the ViewerService from the DocsService. 

Can see the nicety of potentially removing the TreeService dependency from DocsService and passing in the Meta ID directly and do the conversion in the Docs component if you're concerned about that being coupled.